### PR TITLE
Feature/tifsupport

### PR DESCRIPTION
**Pull request by jraval**
# Description

A function for preprocessing steps was created so the user can choose the type and order of preprocessing as well as adding or eliminating steps within the function.  Previously a line of code had to be added for each step of preprocessing.  That is no longer the case.   A loom video will also be created shortly.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [ ] Yes
- [x] No
- [ ] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

**Review by nlittlepoole — CHANGES_REQUESTED** [review]
If you re add these images using git-lfs then you can drop the draft tag and we can probably merge this one

File: data_management/autopreprocess_testing/processing_pipeline.py [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+    "remove_lines": removeLines,
+    "remove_noise": remove_noise,
+    "adaptive_gaussian_thresholding": adaptiveGaussianThresholding,
+    "erode": erode
+}
+
+
+def processing_pipeline(img, queue=None): # Default parameter list will be treated as global
     """
     Takes and image and passes the image through a series of pre-processing steps
```
Comment (bofori20) [inline data_management/autopreprocess_testing/processing_pipeline.py:16]: A little typo to correct 'an'

File: data_management/autopreprocess_testing/test_preprocess_directory.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import preprocess_directory
+import unittest
+FIXTURES_DIRECTORY_PATH = "Fixtures"
+
+class TestGetImages(unittest.TestCase):
```
Comment (bofori20) [inline data_management/autopreprocess_testing/test_preprocess_directory.py:5]: Great to see that you are unit testing.

**Review by bofori20 — COMMENTED** [review]
You should be able to add images to git lfs as suggested. It doesn't look like this is still in draft mode. I suggest you make changes to get it merged as this is a good work.

File: data_management/autopreprocess_testing/processing_pipeline.py [hunk 3]
```diff
@@ -1,6 +1,17 @@
+# from unittest import case
+# from sklearn import preprocessing
```
Comment (jinjahninjah) [inline data_management/autopreprocess_testing/processing_pipeline.py:2]: Please remove the commented out code.

**Review by jinjahninjah — CHANGES_REQUESTED** [review]
Great work Jay! Looks like you have some commented out code in the processing_pipeline.py, and create a Loom video. Once that change and video are made, it should be ready to ship 👍

File: data_management/autopreprocess_testing/processing_pipeline.py (hunk 3 above)
Comment (jraval) [reply data_management/autopreprocess_testing/processing_pipeline.py:2]: Can I do that with GitHub?  I have a revision on another branch.

Comment (james-caldwell1981) [reply data_management/autopreprocess_testing/processing_pipeline.py:2]: You technically can, but I think that would be bad practice. You could either finish your current commit on that branch and switch branches or stash the changes if the documents have already been added to the timeline.

**Comment by jraval** [comment]
Created an update with another branch that includes Feature/extsupport amended #120 to update the preprocessing pipeline file (named processing_pipeline.py) file and preprocessing directory file(preprocessing_directory.py).  The links are below:

feature/extsupport_amended

https://github.com/BloomTech-Labs/scribble-stadium-ds/pull/120

**Comment by jraval** [comment]
Closed tif/support pull request #115.  Details were given in the last comment.

**Pull request closed without merging**